src/btree/bt_slvg.c

```diff
@@ -157,11 +157,77 @@ static int  __slvg_trk_ovfl(WT_SESSION_IMPL *,
 		const WT_PAGE_HEADER *, uint8_t *, size_t, WT_STUFF *);
 
 /*
- * __wt_bt_salvage --
+ * __slvg_checkpoint --
+ *	Create the post-salvage checkpoint.
+ */
+static int
+__slvg_checkpoint(WT_SESSION_IMPL *session, WT_REF *root)
+{
+	WT_BTREE *btree;
+	WT_CKPT *ckptbase;
+	WT_DATA_HANDLE *dhandle;
+	WT_DECL_RET;
+	char *config;
+
+	btree = S2BT(session);
+	dhandle = session->dhandle;
+	config = NULL;
+
+	/*
+	 * XXX
+	 * The salvage process reads and discards previous checkpoints, so the
+	 * underlying block manager has to ignore any previous checkpoint
+	 * entries when creating a new checkpoint, in other words, we can't use
+	 * the metadata checkpoint list, it has all of those checkpoint listed
+	 * and we don't care about them.  Build a clean checkpoint list and use
+	 * it instead.
+	 *
+	 * Don't first clear the metadata checkpoint list and call the function
+	 * to get a list of checkpoints: a crash between clearing the metadata
+	 * checkpoint list and creating a new checkpoint list would look like a
+	 * create or open of a file without a checkpoint to roll-forward from,
+	 * and the contents of the file would be discarded.
```

Comment:
I think this is fine to point out but it would be good to talk about what the code does rather than what we can't do. I'd prefer to see this comment more concise and then add in `Instead we...`

Reply:
@sueloverso 
> it would be good to talk about what the code does rather than what we can't do.

I think the first part of that comment talks about what we do -- can you me know what you think it's missing?